http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java

```diff
@@ -164,10 +165,12 @@ private void verifyMetricsWereCollected(CompletableFuture<Void> metricsFuture) {
     private void makeRequestListener(Future<Channel> channelFuture) {
         if (channelFuture.isSuccess()) {
             channel = channelFuture.getNow();
-            configureChannel();
-            if (tryConfigurePipeline()) {
-                makeRequest();
-            }
+            NettyUtils.doInEventLoop(channel.eventLoop(), () -> {
```

Comment:
I assume this is just an abundance of caution? We should always be in the event loop for the channel right now, right?

Reply:
Actually no, this is part of the fix as well. I also thought that we would be in the eventloop at this point, but currently this method run as a listener to the Promise that contains the leased channel:
https://github.com/aws/aws-sdk-java-v2/blob/bb47fade74ae919beaebb49e48279d57b76c6efe/http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/NettyRequestExecutor.java#L91-L94

However, the eventloop that the promise calls listeners on isn't necessarily the same as the channel's eventloop.